### ISSUE TITLE
Calling dir() on a record produces a TypeError in Python 3.7

### ISSUE THREAD

**Issue by evfredericksen** [GeospatialPython/pyshp#180]
```
Traceback (most recent call last):
  File "main.py", line 31, in <module>
    main()
  File "main.py", line 23, in main
    print(dir(record))
  File "C:\Users\evan\projects\wfrisk\wfrisk\lib\site-packages\shapefile.py", line 471, in __dir__
    return attrs + (self.__field_positions.values()) # plus field names (random order)
TypeError: can only concatenate list (not "dict_values") to list
```

Additionally it seems to make the most sense for dir() to expose self.__field_positions keys rather than values:
 ```python
attrs = [attr for attr in vars(type(self)) if not attr.startswith('_')]
return attrs + list(self.__field_positions) # plus field names (random order)
```

**Comment by karimbahgat** [comment]
Yes, noticed this same bug, and fixed in the current repo. Could you download latest from GitHub and verify that it solves it?

**Comment by evfredericksen** [comment]
It does. Thanks!

**Comment by karimbahgat** [comment]
Great, marking as solved. See also #176.

**Comment by karimbahgat** [comment]
Included in version 2.1.0
